### DISCUSSION TITLE
Is it possible to run the App.main_loop() on a separate thread (not Pythons main_thread) (using the threading or asyncio module)?

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [beeware/toga#1812] [Q&A]
Is it possible to run the App.main_loop() on a separate thread (not Pythons main_thread) (using the threading or asyncio module)?

**Comment by freakboy3742** [comment]
I haven't tested this, but... maybe. The interaction with the GUI framework on each platform is where things will get interesting. GUI frameworks are especially sensitive to which thread is the "main" thread; I *think* you might be safe as long as you create the thread you're going to use for the GUI before you do any GUI actions at all. Behavior across GUI frameworks will vary, so you'll need to test on all platforms you're intending to deploy to. Mobile platforms will be the greatest concern, as there is critical app logic in the bootstrap binary that will be very sensitive to being the "main" thread. Windows will also be an interesting case, due to the use of Single Thread Apartment mode.

**Comment by proneon267** [comment]
Instead of running the `Toga.App.main_loop()` on a separate thread, run your own blocking functions on separate threads. This will work as expected. `Toga.App.main_loop()` won't be very happy if you run it in a separate thread and will throw errors.
